R/parameter_tuning.R

```
@@ -0,0 +1,218 @@
+  if (is.null(registration_method_names)) {
+    registration_method_names = as.character(seq_along(registration_methods))
+  }
+  # Raise error if the list of names isn't the same length as the list of methods
+  # This should only happen if a list is provided but it is spurious
+  if (length(registration_method_names) != length(registration_methods)) {
+    stop("Number of methods and method names do not match")
+  }
+
+  # Initialize the list of lists for the results
+  # The sub-lists for each registration method are indexed by the name of the method
+  per_method_results = list()
+  for (method_name in registration_method_names) {
+    per_method_results[[method_name]] = vector(mode = "list", length = nrow(all_map_param_configurations))
+  }
+
+  # Iterate over the parameter configurations and conduct the registration for each one
+  for (param_ind in seq_len(nrow(all_map_param_configurations))) {
+    message(paste0("iteration: ", param_ind, " / ", nrow(all_map_param_configurations)))
+    # Get one set of map parameters
+    map_params = all_map_param_configurations[param_ind, ]
+    # Create a simulated map based on these parameters
+    simulated_map = do.call(simulate_tree_maps, map_params)
+
+    # Iterate over the different registration methods
+    for (reg_ind in seq_along(registration_methods)) {
+      # Extract the method and method name
+      registration_method = registration_methods[[reg_ind]]
+      registration_method_name = registration_method_names[[reg_ind]]
+      registration_arguments = per_method_registration_arguments[[reg_ind]]
+
+      # Take the important arguments from the simulated map
+      all_registration_args = simulated_map[cbind("pred", "obs", "obs_bounds")]
+      if (!is.null(per_method_registration_arguments)) {
+        all_registration_args = c(all_registration_args, registration_arguments)
+      }
+      all_registration_args = all_registration_args[!duplicated(names(all_registration_args))]
```

Good call to deduplicate. I believe this would only be necessary if the args `pred`, `obs`, or `obs_bounds` were present in `per_method_registration_arguments`, which I don't think they ever should be? Might it be better to return an error if they are, rather than simply removing them?